tests/e2e/helm_test.go

```diff
@@ -0,0 +1,167 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/yaml"
+)
+
+// HelmReleaseStatus describes the possible states of a Helm release.
+type HelmReleaseStatus string
+
+const (
+	// HelmReleaseDeployed is the Helm release state where the deployment is
+	// successfully applied to the cluster.
+	HelmReleaseDeployed HelmReleaseStatus = "deployed"
+
+	// HelmReleaseFailed is the Helm release state where the deployment
+	// encountered an error and couldn't be applied successfully to the cluster.
+	HelmReleaseFailed HelmReleaseStatus = "failed"
+)
+
+// HelmRelease describes a Helm
+type HelmRelease struct {
+	ReleaseName string            `json:"name" yaml:"name"`
+	Namespace   string            `json:"namespace" yaml:"namespace"`
+	Revision    string            `json:"revision" yaml:"revision"`
+	UpdatedTime string            `json:"updated" yaml:"updated"` // Note: not parsable implicitly.
+	Status      HelmReleaseStatus `json:"status" yaml:"status"`
+	Chart       string            `json:"chart" yaml:"chart"`
+	AppVersion  string            `json:"app_version" yaml:"app_version"`
+}
+
+// installHelmChart deploys a Helm chart to the specified kubectl context and
+// namespace using the specified infos, extra arguments can be any of the helm
+// CLI install flag arguments, flag keys and values must be provided separately.
+func installHelmChart(
+	kubectlOptions *k8s.KubectlOptions,
+	helmRepository string,
+	helmChartNameOrLocalPath string,
+	helmChartVersion string,
+	helmReleaseName string,
+	setValues map[string]string,
+) {
+	By(
+		fmt.Sprintf(
+			"Installing Helm chart %s from %s with version %s by name %s",
+			helmChartNameOrLocalPath, helmRepository, helmChartVersion, helmReleaseName,
+		),
+	)
+
+	fixedArguments := []string{
+		"--create-namespace",
+		"--atomic",
+		"--debug",
+	}
+
+	if helmRepository != "" {
+		fixedArguments = append([]string{"--repo", helmRepository}, fixedArguments...)
+	}
+
+	helm.Install(
+		GinkgoT(),
+		&helm.Options{
+			SetValues:      setValues,
+			KubectlOptions: kubectlOptions,
+			Version:        helmChartVersion,
+			ExtraArgs: map[string][]string{
+				"install": fixedArguments,
+			},
+		},
+		helmChartNameOrLocalPath,
+		helmReleaseName,
+	)
+}
+
+// installHelmChart checks whether the specified named Helm release exists in
+// the provided kubectl context and namespace, logs it if it does and returns or
+// alternatively deploys the Helm chart to the specified kubectl context and
+// namespace using the specified infos, extra arguments can be any of the helm
+// CLI install flag arguments, flag keys and values must be provided separately.
+func installHelmChartIfDoesNotExist(
```

Comment:
I have mixed feelings about doing this, I do understand how it would simplify development, but if we would need a non-idempotent Helm install we would need these as separate functions, but that's not the case right now, I will think about merging them.

Reply:
Fixed in cfedc13.